crates/snp_validation/src/lib.rs

```diff
@@ -287,96 +308,132 @@ impl Snp {
+            "snp guest policy mandatory reserved flag not set"
+        );
+
+        // Enarx doesn't support migration to another machine
+        ensure!(
+            !report.body.policy.flags.contains(PolicyFlags::MigrateMA),
+            "snp guest policy migration flag was set"
+        );
+
+        // Only check major.minor, as there isn't a revision or patch version in the attestation.
+        if report.body.policy.abi_major > report.body.current_major {
+            bail!("snp policy has higher abi major than firmware");
+        } else if report.body.policy.abi_minor > report.body.current_minor {
+            bail!("snp policy has higher abi minor than firmware");
         }
 
         // Check reserved fields
         if report.body.rsvd1 != 0 || report.body.rsvd3 != 0 || report.body.rsvd4 != 0 {
-            return Err(anyhow!("snp report reserved fields were set"));
+            bail!("snp report reserved fields were set");
         }
 
         for value in report.body.rsvd2 {
-            if value != 0 {
-                return Err(anyhow!("snp report reserved fields were set"));
-            }
+            ensure!(value == 0, "snp report reserved fields were set");
         }
 
         for value in report.body.rsvd5 {
-            if value != 0 {
-                return Err(anyhow!("snp report reserved fields were set"));
-            }
+            ensure!(value == 0, "snp report reserved fields were set");
         }
 
         for value in report.body.policy.rsvd {
-            if value != 0 {
-                return Err(anyhow!("snp report policy reserved fields were set"));
-            }
+            ensure!(value == 0, "snp report policy reserved fields were set");
         }
 
         for value in report.body.plat_info.rsvd {
-            if value != 0 {
-                return Err(anyhow!("snp report platform_info reserved fields were set"));
-            }
+            ensure!(
+                value == 0,
+                "snp report platform_info reserved fields were set"
+            );
         }
 
         // Check fields not set by Enarx
-        for value in report.body.author_key_digest {
-            if value != 0 {
-                return Err(anyhow!(
-                    "snp report author_key_digest field not set by Enarx"
-                ));
-            }
+        for value in report.body.host_data {
+            ensure!(value == 0, "snp report host_data field not set by Enarx");
         }
 
-        for value in report.body.host_data {
-            if value != 0 {
-                return Err(anyhow!("snp report host_data field not set by Enarx"));
-            }
+        ensure!(report.body.vmpl == 0, "snp report vmpl field invalid value");
+
+        // Check field set by Enarx
+        for value in report.body.report_id_ma {
+            ensure!(
+                value == 255,
+                "snp report report_id_ma field not the value set by Enarx"
+            );
         }
```

Comment:
There's no need to iterate a fixed-size byte array use `==` directly. (Done in the suggested diff)

Reply:
This field is `[u8; 32]`, do you really want a literal array of 32x 255 values?

Reply:
Yes, see diff I sent you earlier

Reply:
no need to check this field... contents is unspecified as no migration agent is in use. this field is not set by us.

Reply:
Even better, we can just drop this check

Reply:
Resolved via your diff

Reply:
How is it acceptable to make sure a field we don't set isn't changed? If that's how it should be done, then why check that the reserved fields aren't zero, since they also aren't set by us?

Nathaniel (I think) wants to be thorough and ensure everything is 100% kosher as best we can.

Reply:
It's just not documented as being 255. Next year it could be 254. Where did you read the specification of 255?

Reply:
The documentation specifies, that the reserved fields must be 0.

Did you read the spec?

Reply:
I did read the spec, and I know the reserved fields must be zero. My questioning has to do with validating fields not set by Enarx.

As to the migration agent field being 255, I don't recall where I saw that, but I do recall working with Nathaniel on it about 7 months ago, which is when this check was added to the code, and it's been the same since.

Reply:
Removed